### PR TITLE
[goldfinch-membership] goldfinch-membership strategy

### PR DESCRIPTION
**Pull request by sanjayprabhu**
New strategies for [goldfinch](https://github.com/goldfinch-eng/mono) protocol

See also https://github.com/snapshot-labs/snapshot-strategies/pull/497

Changes proposed in this pull request:
- A goldfinch membership strategy that checks for eligibility via a UID or the goldfinch golist

### CODE REVIEW

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Hi @sanjayprabhu  Would be great if you can move goldfinch-voting power strategy to a new PR, It is much easier to review and integrate that way :)

**Comment by sanjayprabhu** [comment]
@ChaituVR thanks for the feedback. I moved voting-power to https://github.com/snapshot-labs/snapshot-strategies/pull/497

File: src/strategies/goldfinch-membership/index.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { multicall } from '../../utils';
+import _strategies from '..';
+
+export const author = 'blakewest';
```
Comment (ChaituVR) [inline src/strategies/goldfinch-membership/index.ts:4]:
```suggestion
export const author = 'sanjayprabhu';
```

File: src/strategies/goldfinch-membership/index.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  outputs: [
+    {
+      internalType: 'bool',
+      name: '',
+      type: 'bool'
+    }
+  ],
+  stateMutability: 'view',
+  type: 'function'
+};
```
Comment (ChaituVR) [inline src/strategies/goldfinch-membership/index.ts:25]:
```suggestion
const goListAbi = [ 'function goList(address) view returns (bool)' ]
```

File: src/strategies/goldfinch-membership/index.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      address: UID,
+      symbol: 'UID'
+    },
+    snapshot
+  );
+
+  const goListResult: [[boolean]] = await multicall(
+    network,
+    provider,
+    [goListAbi],
```
Comment (ChaituVR) [inline src/strategies/goldfinch-membership/index.ts:57]:
```suggestion
    goListAbi,
```

File: src/strategies/goldfinch-membership/index.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const strategyFn = _strategies['erc1155-all-balances-of'].strategy;
```
Comment (ChaituVR) [inline src/strategies/goldfinch-membership/index.ts:40]: Would be great if you can change this to import just `erc1155-all-balances-of` strategy into this
```
import strategy as erc1155AllBalancesOf from '../erc1155-all-balances-of';
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by sanjayprabhu** [comment]
Thanks, updated both PRs

**Comment by ChaituVR** [comment]
Thanks @sanjayprabhu Looks good

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**